core/vm/contracts.go

```diff
@@ -599,7 +599,7 @@ func (c *transfer) Run(input []byte, caller common.Address, evm *EVM, gas uint64
 	if err != nil {
 		return nil, gas, err
 	}
-	celoGoldAddress, err := GetRegisteredAddressWithEvm(params.GoldTokenRegistryId, evm)
+	celoGoldAddress, err := evm.Context.GetRegisteredAddress(evm, params.GoldTokenRegistryId)
```

Comment:
I think it would be an improvement to move transfer, and all of our extra precompiled contracts out of the `vm` package.

Reply:
yeah, ideally i would want to minimize the changes to vm module and do everything outside of it. But i didn't want to get too greedy on my first refactor of it....

An interesting exercise is to do a diff between 1.9.13 (our current upstream version) and our code... I'm pretty sure there are things that we added in the past and we actually don't use

Reply:
Cool, added an issue here #1475